stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -639,20 +642,33 @@ public void onClose(Status status, Metadata trailers) {
      * Waits until there is a Runnable, then executes it and all queued Runnables after it.
      */
     public void waitAndDrain() throws InterruptedException {
-      Runnable runnable = queue.take();
-      while (runnable != null) {
+      Runnable runnable = poll();
```

Comment:
Does it throw anymore?

Reply:
This line doesn't but `LockSupport.park()` does (that's where the blocking now happens). The net behaviour should be equivalent to before including w.r.t. interruptions.

Reply:
Are you sure?  Interrupt can wake up park, but it will go back to sleep, making interruption a noop.

Reply:
`InterruptedException` is a checked Exception. It's not in the signature of `LockSupport.park()` so it's never thrown by it. The the thread is interrupted when `LockSupport.park()` is blocking, `LockSupport.park()` just exits.

Reply:
@dapengzhang0 @carl-mastrangelo apologies you're right of course :) need to check interrupt flag after wake-up like you say and throw explicitly